api/v1beta2/inventory_types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+// Inventory is a record of objects that have been reconciled.
+type Inventory struct {
+	// Entries of Kubernetes objects IDs.
+	Entries []Entry `json:"entries"`
+}
+
+// Entry contains the information necessary to locate a resource within a cluster.
+type Entry struct {
+	// ObjectID is the string representation of a Kubernetes object metadata,
+	// in the format '<namespace>_<name>_<group>_<kind>'.
+	ObjectID string `json:"id"`
+
+	// ObjectVersion is the API version of this  Kubernetes object kind.
+	ObjectVersion string `json:"v"`
+}
```

Comment:
Couple of questions and/or suggestions:

1. While `Inventory` as a name makes sense if you know the context, I am wondering if giving it a prefix which provides a more detailed explanation if you are just browsing through e.g. IDE suggestions would ease consumer's their life.
2. Same applies to `Entry`, which provides even less context.
3. The top doc-block comment uses "resource", while the in-struct comments refers to "Kubernetes object". When I added a bunch of helpers to the `runtime` package, I tried to consistently use "Kubernetes resource object", which is also used throughout the Kubernetes core.
4. There is a double space in the `ObjectVersion` doc block.

Suggestion based on the above:

```go
// Inventory contains a list of Kubernetes resource object references that have been applied by a Kustomization.
// Suggestion(hidde): ResourceInventory
type Inventory struct {
	// Entries of Kubernetes resource object references.
	Entries []ResourceRef `json:"entries"`
}

// ResourceRef contains the information necessary to locate a Kubernetes resource object within a cluster.
type ResourceRef struct {
	// ID is the string representation of the Kubernetes resource object's metadata,
	// in the format '<namespace>_<name>_<group>_<kind>'.
	ID string `json:"id"`

	// Version is the API version of the Kubernetes resource object's kind.
	Version string `json:"v"`
}

Reply:
I've updated the inventory according to your suggestions. Thanks!